### PR TITLE
go refactor 2.

### PR DESCRIPTION
**Pull request by vponline**
This applies more PR suggestions to #5 from [here](https://github.com/api3dao/promise-utils/pull/5#pullrequestreview-908585501)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -63,7 +72,7 @@ The return value from the promise utils functions works very well with TypeScrip
 
 The full `promise-utils` API consists of the following functions:
 
-- `go(asyncFn)` - Executes the `asyncFn` and returns a response of type `GoResult`
+- `go(asyncFn, options)` - Executes the `asyncFn` and returns a response of type `GoResult`
 - `goSync(fn)` - Executes the `fn` and returns a response of type `GoResult`
```
Comment (Siegrift) [inline README.md:76]: Mhh, I wonder if goSync should not accept the `PromiseOptions` as well - but I do not see a good enough use for that so let's NOT do that.

File: README.md [hunk 2]
```diff
@@ -76,6 +85,13 @@ and the following Typescript types:
 - `GoResult<T> = { data: T; success: true }`
 - `GoResultSuccess<E extends Error = Error> = { error: E; success: false }`
 - `GoResultError<T, E extends Error = Error> = GoResultSuccess<T> | GoResultError<E>`
+- `PromiseOptions { readonly retries?: number; readonly retryDelayMs?: number; readonly timeoutMs?: number; }`
```
Comment (Siegrift) [inline README.md:88]: Nit: I wonder about the PromiseOptions name, but I do not have a better idea...

**Review by Siegrift — APPROVED** [review]
👍 Thanks

File: README.md (hunk 2 above)
Comment (vponline) [reply README.md:88]: `RetryOptions`? :smile:

File: README.md [hunk 3]
```diff
@@ -76,6 +85,13 @@ and the following Typescript types:
...
+
+The default values for `PromiseOptions` are:
+
+```ts
+{ retries: 0, retryDelay: 200, timeoutMs: 0 }
```
Comment (amarthadan) [inline README.md:93]: Maybe we can add that timeout 0 means no timeout. Even though I guess it's quite obvious.

File: README.md (hunk 2 above)
Comment (Siegrift) [reply README.md:88]: But it also does timeout so retry options is not accurate. Maybe just `Options` but I don't mind this way too.

**Review by Siegrift — APPROVED** [review]
💯 LGTM. Feel free to merge and we can release this.

**Pull request merged**